analytics_data_api/management/commands/generate_data.py

```diff
@@ -0,0 +1,339 @@
+
+    def generate_video_data(self, course_id, video_id, module_id, database):
+        users_at_start = 1234
+        models.Video.objects.using(database).create(course_id=course_id, pipeline_video_id=video_id,
+                                                    encoded_module_id=module_id, duration=500, segment_length=5,
+                                                    users_at_start=users_at_start,
+                                                    users_at_end=random.randint(100, users_at_start))
+
+    def generate_learner_engagement_data(self, course_id, username, start_date, end_date, database, max_value=100):
+        logger.info("Deleting learner engagement module data...")
+        models.ModuleEngagement.objects.using(database).all().delete()
+
+        logger.info("Generating learner engagement module data...")
+        current = start_date
+        progress = tqdm(total=(end_date - start_date).days + 1)
+        while current < end_date:
+            current = current + datetime.timedelta(days=1)
+            for metric in engagement_events.INDIVIDUAL_EVENTS:
+                num_events = random.randint(0, max_value)
+                if num_events:
+                    for _ in range(num_events):
+                        count = random.randint(0, max_value / 20)
+                        entity_type = metric.split('_', 1)[0]
+                        event = metric.split('_', 1)[1]
+                        entity_id = f'an-id-{entity_type}-{event}'
+                        models.ModuleEngagement.objects.using(database).create(
+                            course_id=course_id, username=username, date=current,
+                            entity_type=entity_type, entity_id=entity_id, event=event, count=count)
+            progress.update(1)
+        progress.close()
+        logger.info("Done!")
+
+    def generate_learner_engagement_range_data(self, course_id, start_date, end_date, database, max_value=100):
+        logger.info("Deleting engagement range data...")
+        models.ModuleEngagementMetricRanges.objects.using(database).filter(course_id=course_id).delete()
+
+        logger.info("Generating engagement range data...")
+        for event in engagement_events.EVENTS:
+            low_ceil = random.random() * max_value * 0.5
+            models.ModuleEngagementMetricRanges.objects.using(database).create(
+                course_id=course_id, start_date=start_date, end_date=end_date, metric=event,
+                range_type='low', low_value=0, high_value=low_ceil)
+            high_floor = random.random() * max_value * 0.5 + low_ceil
+            models.ModuleEngagementMetricRanges.objects.using(database).create(
+                course_id=course_id, start_date=start_date, end_date=end_date, metric=event,
+                range_type='high', low_value=high_floor, high_value=max_value)
+
+    def generate_tags_distribution_data(self, course_id, database):
```

Comment:
I don't even know if we can see this tags stuff anywhere in insights currently but this is not the time to drop it